src/shared/dropdown-list-v2/dropdown-list.tsx

```diff
@@ -84,6 +89,15 @@ export const DropdownList = <T, V>({
     const searchInputRef = useRef<HTMLInputElement>(null);
     const virtuosoRef = useRef<VirtuosoHandle>(null);
 
+    const hasSelectedMax = useMemo(
+        () =>
+            maxSelected &&
+            listItems &&
+            maxSelected < listItems.length &&
+            maxSelected > 1,
```

Comment:
hmm doesn't `maxSelected > 1` just mean at least 2 items are selected?

if tracking whether user has selected the upper limit, I thought the comparison would be something like `selectedItems.length === maxSelected`